clients/cloud/java/src/main/java/io/confluent/examples/clients/cloud/ProducerExample.java

```diff
@@ -0,0 +1,117 @@
+import org.apache.kafka.clients.admin.AdminClientConfig;
+import org.apache.kafka.clients.admin.NewTopic;
+
+import io.confluent.examples.clients.cloud.model.DataRecord;
+
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.util.concurrent.ExecutionException;
+import org.apache.kafka.common.errors.TopicExistsException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.Properties;
+import java.util.Collections;
+import java.util.Map;
+
+public class ProducerExample {
+
+  // Create topic in Confluent Cloud
+  public static void createTopic(final String topic,
+                          final int partitions,
+                          final int replication,
+                          final Properties cloudConfig) {
+      final NewTopic newTopic = new NewTopic(topic, partitions, (short) replication);
+      try (final AdminClient adminClient = AdminClient.create(cloudConfig)) {
+          adminClient.createTopics(Collections.singletonList(newTopic)).all().get();
+      } catch (final InterruptedException | ExecutionException e) {
+          // Ignore if TopicExistsException, which may be valid if topic exists
+          if (!(e.getCause() instanceof TopicExistsException)) {
+              throw new RuntimeException(e);
+          }
+      }
+  }
+
+  public static void main(String[] args) throws IOException {
+    if (args.length != 2) {
+      System.out.println("Please provide command line arguments: configPath topic");
+      System.exit(1);
+    }
+
+    // Load properties from file
+    Properties props = loadConfig(args[0]);
+
+    // Create topic if needed
+    String topic = args[1];
+    createTopic(topic, 1, 3, props);
+
+    // Add additional properties.
+    props.put(ProducerConfig.ACKS_CONFIG, "all");
+    props.put(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG, "org.apache.kafka.common.serialization.StringSerializer");
+    props.put(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG, "io.confluent.kafka.serializers.KafkaJsonSerializer");
+
+    Producer<String, DataRecord> producer = new KafkaProducer<String, DataRecord>(props);
+
+    // Produce sample data
+    final Long numMessages = 10L;
+    for (Long i = 0L; i < numMessages; i++) {
+      String key = "alice";
+      DataRecord record = new DataRecord(i);
+
+      System.out.printf("Producing record: %s\t%s%n", key, record);
+      producer.send(new ProducerRecord<String, DataRecord>(topic, key, record), new Callback() {
+          @Override
+          public void onCompletion(RecordMetadata m, Exception e) {
+            if (e != null) {
+              e.printStackTrace();
+            } else {
+              System.out.printf("Produced record to topic %s partition [%d] @ offset %d%n", m.topic(), m.partition(), m.offset());
+            }
+          }
+      });
+    }
+
+    producer.flush();
+
+    System.out.printf("10 messages were produced to topic %s%n", topic);
```

Comment:
nit: probably fine for the example, but technically some of the produce attempts may have failed.

Reply:
@edenhill also had made a comment along those lines when the previous verbiage included the word `successfully` -- which has since been removed :)  I'm going to leave the example as-is, we can revisit later if needed